eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/Mutators.scala

```diff
@@ -0,0 +1,50 @@
+package fr.acinq.eclair.wire.internal.channel
+
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Transaction}
+import fr.acinq.eclair.channel.{ClosingFeerates, Commitments, RealScidStatus, ShortIds}
+import fr.acinq.eclair.wire.protocol._
+import fr.acinq.eclair.{Alias, BlockHeight, RealShortChannelId}
+import shapeless.{::, HNil}
+
+/**
+ * This class contains mutators for [[shapeless.HList]] that are meant to be stacked to account for incremental minor
+ * changes to classes.
+ *
+ * Usage:
+ * {{{
+ *   def codec: Codec[MyClass] = (
+ *   // original codec
+ *   ...
+ *   )
+ *   .map(mutator1)
+ *   .map(mutator2)
+ *   .map(mutator3)
+ *   .decodeOnly
+ *   .as[MyClass]
+ * }}}
+ */
+object Mutators {
+
+  object AddLocalAlias {
+    def genDeterministicAlias(channelId: ByteVector32): Alias = Alias(channelId.take(7).toLong(signed = false))
```

Comment:
As seen offline, we should take the last bytes to avoid collisions when batching opens in a single transaction.

Reply:
Done in [d900265](https://github.com/ACINQ/eclair/pull/2316/commits/d9002656051f59e41d676f9ae0e22077d56d4fb9).